### DISCUSSION TITLE
[Routing] How to Remove Unnecessary Slashes in Symfony Routing Generated Links

### DISCUSSION THREAD

**Discussion by yr4ik** [symfony/symfony#59201] [Q&A]
Symfony routing 5.4.48 (Without framework)

```
$routes = new RouteCollection();

$routes->add('product_list', new Route(
    '/products/{_locale}/{category}/{brand}.html',
    [
        '_locale' => '',   // Default value _locale
        'brand' => ''     // Default value brand
    ],
    [
        '_locale' => 'en|ua|',       // regexp _locale (ua en or none)
        'category' => '[a-z0-9-]+',   // required category
        'brand' => '[a-z0-9-]*'      // brand (allow empty)
    ]
));

$context = new RequestContext('/');

$generator = new UrlGenerator($routes, $context);

echo $generator->generate('product_list', [
    '_locale' => null,
    'category' => 'electronics',
    'brand' => null,
]);
```

The result of this code will be /products//electronics/.html
But it should be /products/electronics.html
How can I remove slashes?

**Comment by stof** [comment]
you will need separate routes for that. The routing component will not allow omitting a placeholder and its previous delimiter entirely if it is followed by some other parts in the URL. `/products/electronics.html` will not match your `product_list` route (and the URL Generator will not want to generate that, as the goal is that the URL it generates can be matched properly by the URL matcher for a request coming to that URL).
